server/dns_server.mli

```diff
@@ -141,13 +144,13 @@ module Primary : sig
      [packet], returning new state, an answer, and potentially notify packets to
      secondary name servers. *)
 
-  val handle_buf : s -> Ptime.t -> int64 -> proto ->
+  val handle_buf : ?packet_callback:packet_callback -> s -> Ptime.t -> int64 -> proto ->
     Ipaddr.t -> int -> Cstruct.t ->
     s * Cstruct.t list * (Ipaddr.t * Cstruct.t list) list *
     [ `Notify of Soa.t option | `Signed_notify of Soa.t option | `Keep ] option *
     [ `raw ] Domain_name.t option
-  (** [handle_buf s now ts proto src src_port buffer] decodes the [buffer],
-     processes the DNS frame using {!handle_packet}, and encodes the reply.
+  (** [handle_buf ~packet_callback s now ts proto src src_port buffer] decodes the
+     [buffer], processes the DNS frame using {!handle_packet}, and encodes the reply.
```

Comment:
in the docstring, the packet_callback should be mentioned and its semantics. but this is best done after we agree on the type of the callback.